### ISSUE TITLE
move all InitErgmTerm functions to one file?

### ISSUE THREAD

**Issue by chad-klumb** [statnet/ergm#85]
talking about the unweighted ones here

currently most (like 90+%) reside in InitErgmTerm.R

the rest are spread out over several files, most of which have a filename starting with "InitErgmTerm", but at least one of which doesn't (InitErgm.bipartite.R)

some of these (including those in InitErgm.bipartite.R) appear in the ergm-terms.Rd file

IMO it would be good to get all the terms into one file, or perhaps two files, one for terms we intend for people to use, and another with terms used for "testing purposes only"

I am not sure if there was a reason why they were split up in the first place, but it's easy to forget the scattered terms when working with InitErgmTerm.R

**Comment by sgoodreau** [comment]
Streamlining and rationalizing the organization of the terms is definitely an important goal. Last I remember the discussion we were pondering different options, including moving nearly all of the terms to a separate package - @martinamorris in particular seemed keen on this idea.  So I think the issue here is part of a bigger discussion. I keep going around in circles in my own head about what option is best.

(Note: I am out of the loop given that I'm far away and not attending F2F meetings, so if you all had this discussion yesterday and this is the outcome, then forgive me!)

**Comment by chad-klumb** [comment]
I think there is still a broader discussion going on.

This issue was meant to address the immediate "problem" that terms are kind of scattered within the ergm package and it's easy to forget/miss something when making term changes.

Moving terms to one file would be quick and easy now, and it wouldn't preclude going in other directions in the longer term.

**Comment by sgoodreau** [comment]
Got it.  Makes sense to me then.

**Comment by martinamorris** [comment]
agreed -- @krivit do you have any objection to moving all terms to one file?

**Comment by krivit** [comment]
The downsides that I see are:
1. It will diminish Git's ability to follow the term's development history.
2. We would need to make sure that the updates propagate to `dev` correctly---so even if we do it, we might want to hold off until `dev` is finally merged into `master`.

Also, maybe it's because I use the command line and therefore have the power of `grep` at my fingertips at all times, I am not seeing that big an advantage: term files have a pretty standard naming scheme, and it's pretty straightforward to iterate through all term files.

I think that if anything, it should be a part of a bigger reorganisation of how we manage and index terms, possibly with that `Roxygen` model idea.

**Comment by chad-klumb** [comment]
I use grep too, but I see no rhyme or reason to the way terms are currently scattered across files within ergm.

Moving them to one file at the same time we port the documentation to roxygen could make sense anyway, though.

**Comment by martinamorris** [comment]
sounds like a plan.

**Comment by krivit** [comment]
@martinamorris , I wouldn't make this a milestone, since as far as I can tell, there is no urgency to this. If and when we remake the documentation system, we might want to consider rearranging the term initialisers ranging from having one file per term to all terms in the same file, but until then, it'll just make merges more complicated.

**Comment by martinamorris** [comment]
Ok.  Is there a downside to keeping this on our radar for the next package release?

**Comment by krivit** [comment]
No real downside that I can think of. I am not arguing for closing the ticket.

**Comment by krivit** [comment]
This'll have to wait for the Great Merge.

**Comment by krivit** [comment]
Let's hold off on this while #274 is being worked on.

**Comment by krivit** [comment]
Anyone want to pick up this discussion again?
